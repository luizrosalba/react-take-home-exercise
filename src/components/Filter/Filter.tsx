import Button from '../Buttons/Button'
type Filter = {
    setFilter: () => void
}
export default function Filter({setFilter}: Filter) {
  return (
    <>
            <Button 
              filter="all" 
              title={"All"} 
              setFilter={setFilter}
            /> 
            <Button 
              filter="completed" 
              title={"Completed"} 
              setFilter={setFilter}
            /> 
            <Button 
              filter="pending" 
              title={"Pending"} 
              setFilter={setFilter}
            /> 
            </>
  )
}
