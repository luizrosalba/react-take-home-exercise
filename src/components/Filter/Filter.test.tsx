import renderer from "react-test-renderer";
import Filter from "./Filter"

describe('Filter Snapshot', () => {  

    it("Matches DOM Snapshot", () => {
        const filterMock = jest.fn()
        const domTree = renderer.create(<Filter setFilter={filterMock}/>).toJSON();
        expect(domTree).toMatchSnapshot();
      });
  
   
  
  });