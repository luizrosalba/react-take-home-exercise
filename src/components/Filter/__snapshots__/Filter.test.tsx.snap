// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Snapshot Matches DOM Snapshot 1`] = `
[
  <button
    className="text-gray-700"
    onClick={[Function]}
  >
    All
  </button>,
  <button
    className="text-gray-700"
    onClick={[Function]}
  >
    Completed
  </button>,
  <button
    className="text-gray-700"
    onClick={[Function]}
  >
    Pending
  </button>,
]
`;
